promql/parse.go

```diff
@@ -1074,12 +1074,15 @@ func (p *parser) checkType(node Node) (typ model.ValueType) {
 
 	case *Call:
 		nargs := len(n.Func.ArgTypes)
-		if na := nargs - n.Func.OptionalArgs; na > len(n.Args) {
+		if na := nargs - n.Func.OptionalArgs; !n.Func.Variadic && na > len(n.Args) {
 			p.errorf("expected at least %d argument(s) in call to %q, got %d", na, n.Func.Name, len(n.Args))
 		}
-		if nargs < len(n.Args) {
+		if !n.Func.Variadic && nargs < len(n.Args) {
 			p.errorf("expected at most %d argument(s) in call to %q, got %d", nargs, n.Func.Name, len(n.Args))
 		}
+		if n.Func.Variadic && nargs > len(n.Args)+1 {
```

Comment:
Variadic is generally associated with 0 or more, not 1 or more. We shouldn't diverge from those well-known semantics. This also allows us to accommodate the current `OptionalArgs` cases, which I would still prefer to be replaced by this.